src/aks-preview/azext_aks_preview/agentpool_decorator.py

```diff
@@ -427,6 +485,19 @@ def update_custom_ca_trust(self, agentpool: AgentPool) -> AgentPool:
             agentpool.enable_custom_ca_trust = False
         return agentpool
 
+    def update_network_profile(self, agentpool: AgentPool) -> AgentPool:
+        self._ensure_agentpool(agentpool)
+
+        asg_ids = self.context.get_asg_ids()
+        allowed_host_ports = self.context.get_allowed_host_ports()
+        if asg_ids or allowed_host_ports:
+            agentpool.network_profile = self.models.AgentPoolNetworkProfile()
```

Comment:
Would rp handle backfill in case user only wants to update one of them and keep the other one?

Reply:
yes, rp will backfill the value we did not provide.